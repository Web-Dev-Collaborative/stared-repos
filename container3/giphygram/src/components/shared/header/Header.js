import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import SearchFormContainerConnected from '../../../containers/searchFormContainer/SearchFormContainer';
import './header.css';

class HeaderRaw extends React.Component {
  render() {
    return (
      <div>
        <h1 className="text-center italic mt-5 mb-5 logo">
          <Link to="/" className="text-dark">
            GiphyGram
          </Link>
        </h1>
        <div className="mb-3">
          <SearchFormContainerConnected />
        </div>
      </div>
    );
  }
}

export const Header = withRouter(HeaderRaw);
