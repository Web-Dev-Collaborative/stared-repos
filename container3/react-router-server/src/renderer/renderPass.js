import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
