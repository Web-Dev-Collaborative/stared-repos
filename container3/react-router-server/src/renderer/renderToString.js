import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
