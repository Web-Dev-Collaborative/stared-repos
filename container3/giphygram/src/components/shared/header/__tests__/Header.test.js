import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Header } from '../Header';

jest.mock(
  '../../../../containers/searchFormContainer/SearchFormContainer',
  () => 'SearchFormContainerConnected',
);

describe('Header', () => {
  it('should be rendered correctly', () => {
    const component = renderer.create((
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
