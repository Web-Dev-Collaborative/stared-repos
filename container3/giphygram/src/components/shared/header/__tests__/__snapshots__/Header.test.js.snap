// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered correctly 1`] = `
<div>
  <h1
    className="text-center italic mt-5 mb-5 logo"
  >
    <a
      className="text-dark"
      href="/"
      onClick={[Function]}
    >
      GiphyGram
    </a>
  </h1>
  <div
    className="mb-3"
  >
    <SearchFormContainerConnected />
  </div>
</div>
`;
